kernel/src/engine/arrow_conversion.rs

```diff
@@ -32,7 +34,10 @@ impl TryFrom<&StructField> for ArrowField {
         let metadata = f
             .metadata()
             .iter()
-            .map(|(key, val)| Ok((key.clone(), serde_json::to_string(val)?)))
+            .map(|(key, val)| match &val {
+                &MetadataValue::String(val) => Ok((key.clone(), val.clone())),
+                _ => Ok((key.clone(), serde_json::to_string(val)?)),
```

Comment:
Am I understanding correctly that the first match arm is for metadata values that were already strings, and the second is for metadata values of other types, that need to become strings so ArrowField can grok them?

Reply:
Yup! That's what's happening